code/game/machinery/doors/firedoor.dm

```diff
@@ -709,6 +709,8 @@
 		else
 			unbuilt_lock.constructionStep = CONSTRUCTION_NO_CIRCUIT
 			unbuilt_lock.update_integrity(unbuilt_lock.max_integrity * 0.5)
+		if(istype(src, /obj/machinery/door/firedoor/border_only))
+			unbuilt_lock.setDir(dir)
```

Comment:
```suggestion
		unbuilt_lock.setDir(dir)
```
Even if you set a dir for full tile firelocks they won't make a difference visually but if its directional it gets set correctly, so there's no need to type check for `src`

Reply:
Fair enough. 
Initial worry was future-proofing but thinking about it this WOULD be future proofing (if full firelocks got directions then the frames would want to do this)